global/matlab/Compile_All_MATLAB_code.sh

```diff
@@ -113,50 +113,28 @@ get_options()
 }
 
 # ------------------------------------------------------------------------------
-#  Compile MATLAB code used for ICAFIX
+#  Main processing of script.
 # ------------------------------------------------------------------------------
-compile_ICAFIX_code()
+main()
 {
+	get_options "$@"
+	
 	log_Msg "----------------------------------------"
 	log_Msg "Compiling ICAFIX-related MATLAB code"
 	log_Msg "----------------------------------------"
 	${HCPPIPEDIR}/ICAFIX/scripts/Compile_MATLAB_code.sh
-}
 
-# ------------------------------------------------------------------------------
-#  Compile MATLAB code used for MSMAll
-# ------------------------------------------------------------------------------
-compile_MSMALL_code()
-{
 	log_Msg "----------------------------------------"
 	log_Msg "Compiling MSMAll-related MATLAB code"
 	log_Msg "----------------------------------------"
 	${HCPPIPEDIR}/MSMAll/scripts/Compile_MATLAB_code.sh
-}
 
-# ------------------------------------------------------------------------------
-#  Compile MATLAB code used for RestingStateStats
-# ------------------------------------------------------------------------------
-compile_RestingStateStats_code()
-{
 	log_Msg "----------------------------------------"
 	log_Msg "Compiling RestingStateStats-related MATLAB code"
 	log_Msg "----------------------------------------"
 	${HCPPIPEDIR}/RestingStateStats/scripts/Compile_MATLAB_code.sh
 }
 
-# ------------------------------------------------------------------------------
-#  Main processing of script.
-# ------------------------------------------------------------------------------
-main()
-{
-	get_options "$@"
-	
-	compile_ICAFIX_code
-	compile_MSMALL_code
-	compile_RestingStateStats_code
-}
```

Comment:
I would prefer that we not make these changes to the "master" compile script. Having three separate `compile_*_code` functions within `main` makes it very convenient to comment out the sub-pipelines that don't need re-compiling (e.g., if we only changed ICAFIX code, and thus only need to recompile that).

Reply:
1) Scripts already exist to compile only a single pipeline's matlab code, and require no editing, and are co-located with that pipeline's matlab files, which should be more convenient than editing anything.
2) A function that calls only one line of processing code needlessly lengthens a script and makes it look more complicated.
3) You can still comment out only one line to prevent the actual compilation step, and I would be perfectly happy removing the `echo`s entirely, but I have never had to read the script output.
4) If the script should officially support not compiling everything, then that suggests it should have options for this, rather than requiring editing.

Given that the only option actually supported is the help info option, and the scripts it calls do their own checking of environment variables anyway, I would instead be tempted to replace the whole thing with a 5-line script:
```
#!/bin/bash
set -eux
"$HCPPIPEDIR"/ICAFIX/scripts/Compile_MATLAB_code.sh
"$HCPPIPEDIR"/MSMAll/scripts/Compile_MATLAB_code.sh
"$HCPPIPEDIR"/RestingStateStats/scripts/Compile_MATLAB_code.sh
```

Reply:
I see your points, regarding the originally proposed changes.  I'm not a fan though of going with the entirely minimalist approach (just 3 lines), as I think the Usage statement adds valuable context and information.